src/components/RoomView/Loser.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { database } from '../../utils/firebase';
+import { ref, get, onValue } from 'firebase/database';
+import { useSelector } from 'react-redux';
+import { setRoomId } from '../../store/playerSlice';
+interface ReduxState {
+  player: { playerId: string };
+  game: { status: string; loser: string };
+}
+  const Loser: React.FC= () => {
+    const { roomId } = useParams();
+    const playerId= useSelector((state: ReduxState) => state.player.playerId)
+    const gameStatus=useSelector((state: ReduxState)=>state.game.status)
+    const loserRef = ref(database, `rooms/${roomId}/game/loser`);
+    const [isVisible, setIsVisible] = useState(true);
+    const[playerIdArray, setPlayerIdArray]=useState<string[]>([])
+    useEffect(()=>{
+      onValue(loserRef, async (loserSnapshot) => {
+        if (loserSnapshot.exists()) {
+           const teamLoser=loserSnapshot.val();
+           const teamLoserRef = ref(database, `rooms/${roomId}/${teamLoser}/`);
+           get(teamLoserRef).then (async(loserMemberSnapshot)=>{
+            if(loserMemberSnapshot.exists())
+            {const loserMember=loserMemberSnapshot.val()
+              const getPlayerIds=(obj:any):string[] =>{
```

Comment:
We should really avoid using "any" in typescript, could you make the type more specific?

Reply:
I can't find a way to avoid using any here other than defining all the types in the interface, which feels a bit redundant for a helper function. Because we are using a "for..in" in the next couple of lines to loop through the object of team-1/team-2: {spymaster: {playerID: someId, username: name} operator:{firebaseKey{playerID: someId, username: name}}.